### PR TITLE
CodeMirror: Make argument to showHint optional

### PR DESCRIPTION
**Pull request by calebegg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 

https://codemirror.net/doc/manual.html
> Defines editor.showHint, which takes an optional options object

- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## codemirror ([<kbd>unpkg</kbd>](https://unpkg.com/browse/codemirror@latest/))
was missing the following properties:
1. optionHandlers
1. inputStyles
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c77576d2e1310c0d9d2a8e534aba2ad3ab4aa991
</p>

File: types/codemirror/test/addon/hint/show-hint.ts [hunk 1]
```diff
@@ -46,3 +46,5 @@ cm.showHint({
     completeSingle: false,
     hint: asyncHintFunc,
 });
+
+cm.showHint();
```
Comment (peterblazejewicz) [inline types/codemirror/test/addon/hint/show-hint.ts:50]: so can we test all?
```ts
cm.showHint();
cm.showHint({});
cm.showHint({ async: true });
```
(I could be wrong about `async` option btw, just example)

**Review by peterblazejewicz — APPROVED** [review]
LGMT!

File: types/codemirror/test/addon/hint/show-hint.ts (hunk 1 above)
Comment (calebegg) [reply types/codemirror/test/addon/hint/show-hint.ts:50]: cm.showHint({}) is not valid. The other two are already tested.

**Review by peterblazejewicz — APPROVED** [review]

**Review by amcasey — APPROVED** [review]

**Comment by calebegg** [comment]
Ready to merge

**Comment by calebegg** [comment]
Ready to merge.

**Review by ficristo — APPROVED** [review]

**Comment by ficristo** [comment]
Ready to merge

**Pull request merged**